stdlib/LinearAlgebra/src/bidiag.jl

```diff
@@ -541,10 +541,7 @@ function rmul!(B::Bidiagonal, D::Diagonal)
     return B
 end
 
-function check_A_mul_B!_sizes(C, A, B)
-    mA, nA = size(A)
-    mB, nB = size(B)
-    mC, nC = size(C)
+@noinline function check_A_mul_B!_sizes((mC, nC)::NTuple{2,Integer}, (mA, nA)::NTuple{2,Integer}, (mB, nB)::NTuple{2,Integer})
```

Comment:
Does the signature pose a problem for infinite arrays?

Reply:
No, as the size is still a tuple of integers. We may potentially remove the type signature though. I had included it to ensure that the method errors early if one passes arrays as arguments.